fluxcd-community/helm-charts
# add possibilty to add secrets for git; some fixes

Signed-off-by: Daniel Werdermann <daniel.werdermann@gmail.com>


#### What this PR does / why we need it:

* add possibility to create a scret to be used in the gitRepository
  * add unittest for the secret
* change flux-cli check job hook-delete-policy to before-hook-creation,hook-succeeded so it gets deleted and does not spam your namespace
* add possibility to create secret to be use in kustomize controller

#### Checklist

- [x] [DCO](https://github.com/fluxcd-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] helm-docs are updated
- [x] Helm chart is tested
